src/main/java/fi/otavanopisto/kuntaapi/server/persistence/dao/ArchivedIdentifierDAO.java

```diff
@@ -0,0 +1,203 @@
+   * 
+   * @param type identifier type
+   * @param source source
+   * @param sourceId id in source system
+   * @return found identifier or null if non found
+   */
+  public ArchivedIdentifier findByTypeSourceSourceIdAndOrganizationKuntaApiId(String type, String source, String sourceId, String organizationKuntaApiId) {
+    EntityManager entityManager = getEntityManager();
+
+    CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+    CriteriaQuery<ArchivedIdentifier> criteria = criteriaBuilder.createQuery(ArchivedIdentifier.class);
+    Root<ArchivedIdentifier> root = criteria.from(ArchivedIdentifier.class);
+    criteria.select(root);
+    criteria.where(
+      criteriaBuilder.and(
+          criteriaBuilder.equal(root.get(ArchivedIdentifier_.type), type),
+          criteriaBuilder.equal(root.get(ArchivedIdentifier_.source), source),
+          criteriaBuilder.equal(root.get(ArchivedIdentifier_.sourceId), sourceId),
+          organizationKuntaApiId == null 
+            ? criteriaBuilder.isNull(root.get(ArchivedIdentifier_.organizationKuntaApiId)) 
+            : criteriaBuilder.equal(root.get(ArchivedIdentifier_.organizationKuntaApiId), organizationKuntaApiId)
+      )
+    );
+    
+    return getSingleResult(entityManager.createQuery(criteria));
+  }
+  
+  /**
+   * Finds archived identifier by type, kuntaApiId and organizationKuntaApiId
+   * 
+   * @param type identifier type
+   * @param source source
+   * @param sourceId id in source system
+   * @return found identifier or null if non found
+   */
+  public ArchivedIdentifier findByTypeAndKuntaApiIdAndOrganizationKuntaApiId(String type, String kuntaApiId, String organizationKuntaApiId) {
+    EntityManager entityManager = getEntityManager();
+
+    CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+    CriteriaQuery<ArchivedIdentifier> criteria = criteriaBuilder.createQuery(ArchivedIdentifier.class);
+    Root<ArchivedIdentifier> root = criteria.from(ArchivedIdentifier.class);
+    criteria.select(root);
+    criteria.where(
+      criteriaBuilder.and(
+          criteriaBuilder.equal(root.get(ArchivedIdentifier_.type), type),
+          criteriaBuilder.equal(root.get(ArchivedIdentifier_.kuntaApiId), kuntaApiId),
+          organizationKuntaApiId == null 
+            ? criteriaBuilder.isNull(root.get(ArchivedIdentifier_.organizationKuntaApiId)) 
+            : criteriaBuilder.equal(root.get(ArchivedIdentifier_.organizationKuntaApiId), organizationKuntaApiId)
+      )
+    );
+    
+    return getSingleResult(entityManager.createQuery(criteria));
+  }
+//  
+//  
+//
+//  /**
+//   * Finds identifier by type, source and Kunta API id
+//   * 
+//   * @param type identifier type
+//   * @param source source
+//   * @param kuntaApiId Kunta API id 
+//   * @return found identifier or null if non found
+//   */
+//  public ArchivedIdentifier findByTypeSourceAndKuntaApiId(String type, String source, String kuntaApiId) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)